run.py

```diff
@@ -0,0 +1,141 @@
+        class_name = class_name + name.capitalize()
+
+    # Build the full file path to the test as a string
+    file_path = generate_file_path(test_path, test_name + ".py")
+
+    # Load module with test class given the file (test) name and file path
+    module = imp.load_source(test_name, file_path)
+
+    # Create instance of test class
+    test_class = getattr(module, class_name)(browser)
+
+    # Add instance of test class to dictionary
+    tests_loaded[test_name] = test_class
+
+
+def open_file(file_name):
+    """
+    Opens a file in read only mode and returns its content in JSON.
+
+    :param file_name: (strong) The name of the file to be opened
+    :return: (json) Contents of file
+    """
+    try:
+        with open(file_name, 'r') as f:
+            return json.load(f)
+    except Exception as e:
+        print "ERROR: Check your JSON file for syntax errors"
+        print e
+
+
+def generate_file_path(package_path, file_name):
+    """
+    Dynamically generate full path to file, including filename and extension.
+
+    :param package_path: (array) ordered list of packages in path to test file
+    :param file_name: (string) name of the file w/ test, including the extension
+    :return: (string) full path to file, including filename and extension
+    """
+    file_path = ""
+    for package in package_path:
+        file_path = file_path + package + "/"
+    return file_path + file_name
+
+
+def execute_tests(tests):
+    """
+    Executes specified tests.
+
+    :param tests: (json) List of tests to run
+    :return:
+    """
+    for test in tests:
+        # If a test has an enabled field set to true OR does not have an
+        # enabled field, then we assume the test is enabled and run it
+        if test.get("enabled", True):
+            # Run group of tests
+            if test.get("type", "test") == "test-group":
+                # Generate the path to the test group's manifest file
+                filename = generate_file_path(test["test_name"].split("."),
+                                              "manifest.json")
+
+                # Open manifest file for test group
+                tests = open_file(filename)
+
+                # Execute tests in test group
+                execute_tests(tests)
+
+            # Run individual test
+            else:
+                try:
+                    # The test name for an individual test includes the test
+                    # name. We must parse this out to generate the package path.
+                    test_path = test["test_name"].split(".")
+                    test_name = test_path[len(test_path) - 1]
+                    del test_path[-1]
+
+                    # Import module if not already imported
+                    if test_name not in tests_loaded:
+                        import_test(test_path, test_name)
+
+                    # Execute test setup
+                    getattr(tests_loaded[test_name], "setup")()
```

Comment:
Future feature:
Support params for setup & teardown :smile:

Reply:
oOOoooOOO